### PR TITLE
workflows/release-binaries: Enable flang builds on Windows

### LINKED ISSUE #100202: Flang build fails on GitHub Action Windows runners
I'm trying to do a release build with flang on the GitHub Actin Window runners and it is failing with this error:

```
FAILED: bin/bbc.exe 
C:\Windows\system32\cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_exe --intdir=tools\flang\tools\bbc\CMakeFiles\bbc.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\LLVM\bin\lld-link.exe /nologo @CMakeFiles\bbc.rsp  /out:bin\bbc.exe /implib:lib\bbc.lib /pdb:bin\bbc.pdb /version:0.0 /machine:x64 /STACK:10000000 /INCREMENTAL:NO /subsystem:console && cd ."
LINK: command "C:\PROGRA~1\LLVM\bin\lld-link.exe /nologo @CMakeFiles\bbc.rsp /out:bin\bbc.exe /implib:lib\bbc.lib /pdb:bin\bbc.pdb /version:0.0 /machine:x64 /STACK:10000000 /INCREMENTAL:NO /subsystem:console /MANIFEST:EMBED,ID=1" failed (exit code 1) with the following output:

lld-link: error: undefined symbol: __udivti3
>>> referenced by FortranDecimal.lib(binary-to-decimal.cpp.obj):(public: __cdecl Fortran::decimal::BigRadixFloatingPointNumber<64, 16>::BigRadixFloatingPointNumber<64, 16>(class Fortran::decimal::BinaryFloatingPointNumber<64>, enum Fortran::decimal::FortranRounding))
>>> referenced by FortranDecimal.lib(binary-to-decimal.cpp.obj):(public: __cdecl Fortran::decimal::BigRadixFloatingPointNumber<113, 16>::BigRadixFloatingPointNumber<113, 16>(class Fortran::decimal::BinaryFloatingPointNumber<113>, enum Fortran::decimal::FortranRounding))
```

### How To Reproduce

```
cmake -G Ninja -S llvm -B /d/a/llvm-project/llvm-project/build \
         -DLLVM_RELEASE_ENABLE_PGO=OFF -DLLVM_RELEASE_ENABLE_LTO=OFF \
         -C clang/cmake/caches/Release.cmake \
         -DBOOTSTRAP_LLVM_PARALLEL_LINK_JOBS=1
```

### PR DESCRIPTION
**Pull request by tstellar**
Flang for Windows depends on compiler-rt, so we need to enable it for the stage1 builds.  This also fixes failures building the flang tests on macOS.

Fixes #100202.

### CODE REVIEW

**Review by boomanaiden154 — APPROVED** [review]

**Comment by tstellar** [comment]
It looks like this change also fixed some build failures on macOS too.

File: clang/cmake/caches/Release.cmake [hunk 1]
```diff
@@ -47,11 +47,17 @@ set(LLVM_TARGETS_TO_BUILD Native CACHE STRING "")
...
+# stage1.  compiler-rt is also required for building the Flang tests on
+# macOS.
+set(STAGE1_RUNTIMES "compiler-rt")
 
 if (LLVM_RELEASE_ENABLE_PGO)
   list(APPEND STAGE1_PROJECTS "lld")
-  list(APPEND STAGE1_RUNTIMES "compiler-rt")
+  if (NOT "compiler-rt" IN_LIST STAGE1_RUNTIMES)
+    list(APPEND STAGE1_PROJECTS "compiler-rt")
+  endif()
```
Comment (nikic) [inline clang/cmake/caches/Release.cmake:60]: Why is this bit needed?

Comment (tstellar) [reply clang/cmake/caches/Release.cmake:60]: It's probably not needed any more.  I was planning to conditionally add compiler-rt to stage1 targets based on if flang was enabled or not, but now I think that is too complicated.

**Comment by tstellar** [comment]
I'm going to push this tomorrow so it has a chance to land before -rc3.

**Pull request merged**